bin/reth/src/node/cl_events.rs

```diff
@@ -0,0 +1,64 @@
+use futures::Stream;
+use reth_provider::CanonChainTracker;
+use std::{
+    pin::Pin,
+    task::{ready, Context, Poll},
+    time::Duration,
+};
+use tokio::time::Interval;
+
+/// Period of not exchanging transition configurations with consensus client,
+/// after which the warning is issued.
+const NO_TRANSITION_CONFIG_EXCHANGED_PERIOD: Duration = Duration::from_secs(120);
+/// Period of not receiving fork choice updates from consensus client,
+/// after which the warning is issued.
+const NO_FORKCHOICE_UPDATE_RECEIVED_PERIOD: Duration = Duration::from_secs(120);
+
+pub struct ConsensusLayerHealthEvents<CC> {
+    interval: Interval,
+    canon_chain: CC,
+}
+
+impl<CC> ConsensusLayerHealthEvents<CC> {
+    pub fn new(interval: Duration, canon_chain: CC) -> Self {
+        Self { interval: tokio::time::interval(interval), canon_chain }
+    }
+}
+
+impl<CC> Stream for ConsensusLayerHealthEvents<CC>
+where
+    CC: CanonChainTracker + Unpin,
+{
+    type Item = ConsensusLayerHealthEvent;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = self.get_mut();
+
+        ready!(this.interval.poll_tick(cx));
```

Comment:
I recently learned that poll_tick only promises to emit a wakeup when it returns Poll::Pending. If it returned Poll::Ready, then it will not wake you up. You need to call poll_tick again.

so this can be put in a loop

Reply:
where did u read that? interesting

Reply:
function docs, there was also a recent bug fix that actually enforced this properly

Reply:
interesting, I see a bunch of other places in the codebase when we use it without the second call. Should we fix it?

Reply:
good point, but depends, I believe this is only an issue if there's nothing else that would wake up the task.
This is also not the case if the surrounding `poll` function returns Ready because then it will be called again, but in this case it's possible that we return Pending if the interval returned Ready, so there's no waker registered